rskj-core/src/main/java/co/rsk/cli/migration/UnitrieMigrationTool.java

```diff
@@ -0,0 +1,396 @@
+            }
+        }
+
+        if (code != null) {
+            if (!initialized) {
+                currentRepository.setupContract(accountAddress);
+            }
+            if (!Arrays.equals(accountCodeHash, Keccak256Helper.keccak256(code))) {
+                // mati-fix (ref: org.ethereum.db.DetailsDataStore#get)
+                code = orchidContractsStorage.get(accountCodeHash);
+            }
+            currentRepository.saveCode(accountAddress, code);
+        }
+    }
+
+    private static Trie orchidTrieDeserialize(byte[] bytes) {
+        final int keccakSize = Keccak256Helper.DEFAULT_SIZE_BYTES;
+        int expectedSize = Short.BYTES + keccakSize;
+        if (expectedSize > bytes.length) {
+            throw new IllegalArgumentException(
+                    String.format("Expected size is: %d actual size is %d", expectedSize, bytes.length));
+        }
+
+        byte[] root = Arrays.copyOfRange(bytes, Short.BYTES, expectedSize);
+        TrieStore store = orchidTrieStoreDeserialize(bytes, expectedSize, new HashMapDB());
+
+        Trie newTrie = store.retrieve(root);
+
+        if (newTrie == null) {
+            throw new IllegalArgumentException(String.format("Deserialized storage doesn't contain expected trie: %s", Hex.toHexString(root)));
+        }
+
+        return newTrie;
+    }
+
+    private static TrieStore orchidTrieStoreDeserialize(byte[] bytes, int offset, KeyValueDataSource ds) {
+        int current = offset;
+        current += Short.BYTES; // version
+
+        int nkeys = readInt(bytes, current);
+        current += Integer.BYTES;
+
+        for (int k = 0; k < nkeys; k++) {
+            int lkey = readInt(bytes, current);
+            current += Integer.BYTES;
+            if (lkey > bytes.length - current) {
+                throw new IllegalArgumentException(String.format(
+                        "Left bytes are too short for key expected:%d actual:%d total:%d",
+                        lkey, bytes.length - current, bytes.length));
+            }
+            byte[] key = Arrays.copyOfRange(bytes, current, current + lkey);
+            current += lkey;
+
+            int lvalue = readInt(bytes, current);
+            current += Integer.BYTES;
+            if (lvalue > bytes.length - current) {
+                throw new IllegalArgumentException(String.format(
+                        "Left bytes are too short for value expected:%d actual:%d total:%d",
+                        lvalue, bytes.length - current, bytes.length));
+            }
+            byte[] value = Arrays.copyOfRange(bytes, current, current + lvalue);
+            current += lvalue;
+            ds.put(key, value);
+        }
+
+        return new TrieStoreImpl(ds);
+    }
+
+    // this methods reads a int as dataInputStream + byteArrayInputStream
+    private static int readInt(byte[] bytes, int position) {
+        final int LAST_BYTE_ONLY_MASK = 0x000000ff;
```

Comment:
We removed the mask from readShort implementation :smile:

Reply:
Nice catch!